src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -52,6 +52,19 @@ internal static class Utf8Encoder
         /// <returns>True if the input buffer was fully encoded into the output buffer, otherwise false.</returns>
         public static unsafe bool TryDecode(ReadOnlySpan<byte> utf8, Span<char> utf16, out int bytesConsumed, out int charactersWritten)
         {
+            if (utf8.IsEmpty)
```

Comment:
It should not be needed here. The rest of the implementation seems to handle it fine.